### ISSUE TITLE
Add random songs (or artists, or albums) is not random at all

### ISSUE THREAD

**Issue by cronokun** [arybczak/ncmpcpp#120]
When I add, say, 20 random songs to the playlist (by pressing ``` key), then clear playlist, exit, open ncmpcpp again and 20 random songs – they're exactly the same 20 random songs as before. Event if I do this again, but now add 30 random songs – first 20 are still the same as before.

ncmpcpp version: 0.7.3
OS: Mac OS X 10.11.4 15E65

**Comment by jgrar** [comment]
I can't reproduce this in latest git. Could you try with a more recent version?

**Comment by cronokun** [comment]
Installed latest dev version, still reproducible. If it matters, I installed ncmpcpp via homebrew.

```
ncmpcpp 0.8_dev

optional screens compiled-in:
 - tag editor
 - tiny tag editor
 - artist info

encoding detection: enabled
built with support for: curl ncurses taglib unicode
```

**Comment by ianmcxa** [comment]
Cannot reproduce on 0.7.4-2 linux.

**Comment by Hamuko** [comment]
Having the same issue with macOS 10.12 Sierra and 0.8_dev build based on e34221f74d1cfb6140cd40bf57dffeb13e8a5408. Same behavior happened on 0.7.4.

Here's a video showcasing the utter lack of randomness: https://0x0.st/1GA.webm

**Comment by Hamuko** [comment]
I started poking around the code to see what causes a non-random random function, and I discovered that the randomness in the feature comes from `std::random_shuffle(b, e)`. After doing a couple of Google searches about the function, I reached the conclusion that the randomness is actually defined by the implementation, so it might not use the random seed initialized at the start of execution.

I made a commit Hamuko/ncmpcpp@57017285559e59abdeea90343460b1c0225b706e that uses `std::shuffle(b, e, g)` to provide a mt19937 random number generator to the shuffle function explicitly as the third argument. With this change, I can no longer get the same songs at random every time I add tracks after opening the program.

My C++ might not be that elegant though, since I picked up writing the language ~10 hours ago.
